src/lib.rs

```diff
@@ -143,3 +143,62 @@ where
         Some(next_record.record)
     }
 }
+
+#[cfg(test)]
```

Comment:
Would prefer if we could put this test in a separate

```
#[cfg(test)]
mod tests;
```

with a separate file